From 28b03a9a338dc24b002413e880222fe1d49f77f5 Mon Sep 17 00:00:00 2001
From: Michael Dickens <michael.dickens@ettus.com>
Date: Sat, 1 Apr 2017 15:24:46 -0400
Subject: [PATCH 11/11] Use 'powf' to match variables and avoid implicit type
 converstion. Makes some older compilers happy, allowing 'make test' to pass.

---
 kernels/volk/volk_32f_x2_pow_32f.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernels/volk/volk_32f_x2_pow_32f.h b/kernels/volk/volk_32f_x2_pow_32f.h
index 58fecb6..a8cb2e1 100644
--- a/kernels/volk/volk_32f_x2_pow_32f.h
+++ b/kernels/volk/volk_32f_x2_pow_32f.h
@@ -190,7 +190,7 @@ volk_32f_x2_pow_32f_a_sse4_1(float* cVector, const float* bVector,
 
   number = quarterPoints * 4;
   for(;number < num_points; number++){
-    *cPtr++ = pow(*aPtr++, *bPtr++);
+    *cPtr++ = powf(*aPtr++, *bPtr++);
   }
 }
 
@@ -215,7 +215,7 @@ volk_32f_x2_pow_32f_generic(float* cVector, const float* bVector,
   unsigned int number = 0;
 
   for(number = 0; number < num_points; number++){
-    *cPtr++ = pow(*aPtr++, *bPtr++);
+    *cPtr++ = powf(*aPtr++, *bPtr++);
   }
 }
 #endif /* LV_HAVE_GENERIC */
@@ -326,7 +326,7 @@ volk_32f_x2_pow_32f_u_sse4_1(float* cVector, const float* bVector,
 
   number = quarterPoints * 4;
   for(;number < num_points; number++){
-    *cPtr++ = pow(*aPtr++, *bPtr++);
+    *cPtr++ = powf(*aPtr++, *bPtr++);
   }
 }
 
-- 
2.11.0

